### DISCUSSION TITLE
Sbatch gives back "not enough slots"

### DISCUSSION THREAD

**Discussion by GeneralStarbound** [idaholab/moose#21129] [Q&A General] [answered]
When I submit my slurm file for MOOSE to analyze, but when I get it back, I got an error log saying that my file took up too many slots.  I'd only requested 8.  Should I request fewer, or is this a problem with my program.  I'm almost done with a project, just need this to get run.

**Comment by mangerij** [comment]
This seems to be an issue with your SLURM Scheduler and probably your allocation (i.e. not MOOSE). You probably should contact your HPC admin.

**Reply by GiudGiud** [reply]
what does your job submission script look like? We have people familiar with slurm so we may be able to pick up an error

**Comment by ngrilli** [comment, marked as answer]
Dear @GeneralStarbound ,

I have never got that specific error, but if that helps, I share here a typical submit file that I use with SLURM:


#!/bin/bash

#SBATCH --job-name=my_job
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=24
#SBATCH --cpus-per-task=1
#SBATCH --time=25:00:00
#SBATCH --mem-per-cpu=12G

module load tools/git/2.22.0
module load lang/python/anaconda/3.8.5-2021-Jupyter
module load tools/cmake/3.14.2
module add lang/cuda/10.1.105

source ~/moose-environment.sh

srun -n $SLURM_NTASKS --cpu-bind=cores --mpi=pmi2 ./c_pfor_am-opt -i my_input_file.i



Hope this can help you,
Nicolò
